checks.d/consul.py

```diff
@@ -122,23 +122,29 @@ def _check_for_leader_change(self, instance):
             self.log.info(('Leader change from {0} to {1}. Sending new leader event').format(
                 self._last_known_leader, leader))
 
-            self.event({
-                "timestamp": int(datetime.now().strftime("%s")),
-                "event_type": "consul.new_leader",
-                "source_type_name": self.SOURCE_TYPE_NAME,
-                "msg_title": "New Consul Leader Elected in consul_datacenter:{0}".format(agent_dc),
-                "aggregation_key": "consul.new_leader",
-                "msg_text": "The Node at {0} is the new leader of the consul datacenter {1}".format(
-                    leader,
-                    agent_dc
-                ),
-                "tags": ["prev_consul_leader:{0}".format(self._last_known_leader),
-                         "curr_consul_leader:{0}".format(leader),
-                         "consul_datacenter:{0}".format(agent_dc)]
-            })
+            self._emit_leader_change(instance, leader)
 
         self._last_known_leader = leader
 
+    def _emit_leader_change(self, instance, leader):
+        agent_dc = self._get_agent_datacenter(instance)
+
+        tags = ["curr_consul_leader:{0}".format(leader), "consul_datacenter:{0}".format(agent_dc)]
+        if self._last_known_leader:
+            tags.append("prev_consul_leader:{0}".format(self._last_known_leader))
```

Comment:
As [mentioned above](https://github.com/DataDog/dd-agent/pull/2647/files#r69222053) `self_leader_check` doesn't keep track of `_last_known_leader` and thus can't emit it as part of the event.